kvbc/src/Replica.cpp

```diff
@@ -462,11 +465,56 @@ bool Replica::putBlock(const uint64_t blockId, const char *blockData, const uint
           std::to_string(blockId));
     }
   } else {
-    m_kvBlockchain->addRawBlock(rawBlock, blockId);
+    m_kvBlockchain->addRawBlock(rawBlock, blockId, trylinkSTChainFrom);
   }
   return true;
 }
 
+std::future<bool> Replica::putBlockAsync(uint64_t blockId,
+                                         const char *block,
+                                         const uint32_t blockSize,
+                                         bool trylinkSTChainFrom) {
+  static uint64_t callCounter = 0;
+  static constexpr size_t snapshotThresh = 1000;
+
+  auto future = blocksIOWorkersPool_.async(
+      [this](uint64_t blockId, const char *block, const uint32_t blockSize, bool trylinkSTChainFrom) {
+        auto start = std::chrono::steady_clock::now();
+        bool result = false;
+
+        LOG_TRACE(logger, "Job Started: " << KVLOG(blockId, blockSize));
+        try {
+          result = putBlock(blockId, block, blockSize, trylinkSTChainFrom);
+        } catch (const std::runtime_error &ex) {
+          LOG_ERROR(logger, "runtime_error exception:" << ex.what());
+          return false;
+        } catch (const std::exception &ex) {
+          LOG_ERROR(logger, "exception:" << ex.what());
+          return false;
+        } catch (...) {
```

Comment:
We don't catch unknown exceptions and we are not catching exceptions in case putBlock fails, it's a reason for termination. Look at `putBlock()`

Reply:
OK, this is redundant. I'm actually catching an exception at the caller as well. so this is redundant for now..
Will remove..